src/codegen/mod.rs

```diff
@@ -1673,7 +1673,11 @@ impl CodeGenerator for CompInfo {
             attributes.push(attributes::doc(comment));
         }
         if packed && !is_opaque {
-            attributes.push(attributes::repr_list(&["C", "packed"]));
+            let n = layout.map_or(1, |l| l.align);
+            // We must either support `#[repr(packed(N))]` or have N = 1;
+            assert!((n > 1 && ctx.options().rust_features().repr_packed_n) || n == 1);
```

Comment:
I'd just make this assertion `ctx.options().rust_features().repr_packed_n || n == 1`. I think it's clearer, and makes the comment not really needed, since now they read the same :)

Also, maybe just `debug_assert!`? The autogenerated layout tests would catch it in any case.

Reply:
Fixed. I left it as `assert!` because I hate debugging issues that appear during different configurations and the check isn't performance critical.

Reply:
Fair enough :)